eventbus/kafka/sensor/kafka_transaction.go

```diff
@@ -0,0 +1,94 @@
+package kafka
+
+import (
+	"github.com/Shopify/sarama"
+	"go.uber.org/zap"
+)
+
+type KafkaTransaction struct {
+	Logger        *zap.SugaredLogger
+	Producer      sarama.AsyncProducer
+	GroupName     string
+	Topic         string
+	Partition     int32
+	ResetOffset   int64
+	ResetMetadata string
+}
+
+func (t *KafkaTransaction) Commit(session sarama.ConsumerGroupSession, messages []*sarama.ProducerMessage, offset int64, metadata string) error {
+	// No need for a transaction if no messages
+	if len(messages) == 0 {
+		session.MarkOffset(t.Topic, t.Partition, offset, metadata)
```

Comment:
Good catch! It is still very important to commit when there are no messages, this can happen for two reasons.

1) Because the action topic is the final topic, we always bump the offset without producing any messages.
2) The trigger topic can bump the offset without producing a message. Imagine you have have a trigger with  condition `a && b` and consume a stream of `a`'s. Every time we consume a new `a` we want to bump the offset, but we can't produce a message because the trigger condition is not yet satisfied.